modules/oe_content_event/src/Plugin/Validation/Constraint/EventFieldsRequiredValidator.php

```diff
@@ -76,42 +74,6 @@ protected function validateGroupFieldsEmpty(array $fields, Constraint $constrain
     }
   }
 
-  /**
-   * Validate organiser information consistency.
-   *
-   * An organiser can either be a custom string or a reference to a corporate
-   * vocabulary, depending from the value of `oe_event_organiser_is_internal`.
-   *
-   * This tests that, if one is set, the other is always not, depending
-   * whether the organiser is marked as internal or not.
-   *
-   * @param \Symfony\Component\Validator\Constraint $constraint
-   *   The constraint object.
-   * @param \Drupal\node\NodeInterface $node
-   *   The node object.
-   */
-  protected function validateOrganiserGroupFields(Constraint $constraint, NodeInterface $node) {
-    $violation = NULL;
-    if ($node->get('oe_event_organiser_internal')->isEmpty() && $node->get('oe_event_organiser_name')->isEmpty()) {
-      $violation = $this->context->buildViolation('You have to fill in at least one of the following fields: @internal or @organiser_name', [
-        '@internal' => $node->getFieldDefinition('oe_event_organiser_internal')->getLabel(),
-        '@organiser_name' => $node->getFieldDefinition('oe_event_organiser_name')->getLabel(),
-      ]);
-    }
-
-    if ($violation instanceof ConstraintViolationBuilderInterface) {
-      // Highlight empty 'Organiser name' field.
-      (clone $violation)
-        ->atPath('oe_event_organiser_name')
-        ->addViolation();
-
-      // Highlight empty 'Internal organiser' field.
-      $violation
-        ->atPath('oe_event_organiser_internal')
-        ->addViolation();
-    }
-  }
```

Comment:
We cannot remove this, we need to change it, meaning: if they are both empty then we simple ignore the constrain, while if at least one of them is not, then we apply it.

Reply:
The code did not correspond to the description of the function so I updated the description and the code the the new behavior

Reply:
Yesterday I implemented as you wanted :

```
protected function validateOrganiserGroupFields(Constraint $constraint, NodeInterface $node) {
    // Organiser is not required, so all subfields can be empty.
    if ($node->get('oe_event_organiser_internal')->isEmpty() && $node->get('oe_event_organiser_name')->isEmpty()) {
      return;
    }

    $violation = NULL;

    if ($node->get('oe_event_organiser_is_internal')->isEmpty() && !$node->get('oe_event_organiser_internal')->isEmpty()) {
      $violation = $this->context->buildViolation('When @is_internal field is not checked, @internal field have to be empty.', [
        '@is_internal' => $node->getFieldDefinition('oe_event_organiser_is_internal')->getLabel(),
        '@internal' => $node->getFieldDefinition('oe_event_organiser_internal')->getLabel(),
      ]);
    }
    elseif (!$node->get('oe_event_organiser_is_internal')->isEmpty() && !$node->get('oe_event_organiser_name')->isEmpty()) {
      $violation = $this->context->buildViolation('When @is_internal field is checked, @organiser_name field have to be empty.', [
        '@is_internal' => $node->getFieldDefinition('oe_event_organiser_is_internal')->getLabel(),
        '@organiser_name' => $node->getFieldDefinition('oe_event_organiser_name')->getLabel(),
      ]);
    }

    if ($violation instanceof ConstraintViolationBuilderInterface) {
      // Highlight empty 'Organiser name' field.
      (clone $violation)
        ->atPath('oe_event_organiser_name')
        ->addViolation();

      // Highlight empty 'Internal organiser' field.
      $violation
        ->atPath('oe_event_organiser_internal')
        ->addViolation();
    }
  }
```

But today I remove that because I think it is not needed insofar as nothing is required, so nothing have to be checked.
So drone is NOW green

Please @ademarco let me know if I am wrong. thank you

Reply:
I think julien is right here, there is only 1 field shown (depending on the checkbox) so if its filled its filled, if its not its not (basically the essence of optional)

Reply:
We need to make sure only one of the two is filled, depending on the value of the checkbox, but now that those fields are not mandatory anymore, empty values are also admissible. So the following are all valid scenarios:

- "Is internal" TRUE - "internal" EMPTY
- "Is internal" FALSE - "name" EMPTY
- etc.

But the following can never happen:

- "Is internal" TRUE - "name" NOT EMPTY